app/graphql/users/middlewares.js

```diff
@@ -1,14 +1,50 @@
 const logger = require('../../logger');
+const { user: User } = require('../../models');
+const { EMAIL_REGEX, PASSWORD_REGEX } = require('../../constants');
+const errors = require('../../errors');
 
-const createUser = (resolve, root, args) => {
-  logger.info("Middleware for 'createUser' mutation");
-  // Add different actions that you want to be executed before your resolver, i.e: input validation or caching
-  return resolve(root, args);
+const user = (resolve, root, args) => {
+  logger.info("Middleware for 'user' mutation");
+  const errorList = [];
+  const { email, password } = args.userFields;
+  if (!EMAIL_REGEX.test(email)) {
+    logger.error('The email used does not belong to Wolox dominoes');
+    errorList.push({
+      field: 'email',
+      message: 'The email does not belong to the Wolox domains',
+      statusCode: 400
+    });
+  }
+  if (!PASSWORD_REGEX.test(password)) {
+    logger.error('The password must be alphanumeric and should be at least 8 chars long');
+    errorList.push({
```

Comment:
The validation object you are building here is pretty similar to the one you made on the previous `if()`. Maybe you can find a way to make function that builds them in a more generic way? 🤔